### PR TITLE
feat: asyncify pickers

### PR DESCRIPTION
**Pull request by tjdevries**
Working on making pickers async. Will clean up code as I go.

### CODE REVIEW

**Comment by oberblastmeister** [comment]
I am getting this error for the autocommands finder:
```
Error executing vim.schedule lua callback: ...scope.nvim/lua/telescope/previewers/buffer_previewer.lua:562: attempt to index field 'finde
r' (a function value)
```

**Comment by ThePrimeagen** [comment]
As a consumer of this library and an active watcher of such pull requests, I feel that the founder and main contributor to Telescope having linting issues really reduces my confidence in this project.

**Comment by tjdevries** [comment]
Please don't make me turn my profile picture into the leaked photos that have been going around on twitter...

**Comment by elianiva** [comment]
I've tried it out, this is more than just "works for me", but "*works great for me*" :D

**Comment by tjdevries** [comment]
OK, I think I'm ready to merge this.

Could I get some code review / thoughts / anything? I'll wait 24 hours and then merge (so I can fix on Friday on stream if we have any problems)

**Comment by kuator** [comment]
Well to be perfectly honest, in my humble opinion, of course without offending anyone who thinks differently from my point of view, but also by looking into this matter in a different perspective and without being condemning of one's view's and by trying to make it objectified, and by considering each and every one's valid opinion, I honestly believe that I completely forgot what I was going to say

**Comment by kuator** [comment]
> OK, I think I'm ready to merge this.
> 
> Could I get some code review / thoughts / anything? I'll wait 24 hours and then merge (so I can fix on Friday on stream if we have any problems)

Well to be perfectly honest, in my humble opinion, of course without offending anyone who thinks differently from my point of view, but also by looking into this matter in a different perspective and without being condemning of one's view's and by trying to make it objectified, and by considering each and every one's valid opinion, I honestly believe that I completely forgot what I was going to say

File: lua/telescope/finders.lua [hunk 1]
```diff
@@ -298,18 +147,20 @@ finders.new_job = function(command_generator, entry_maker, maximum_results, cwd)
   }
 end
 
----@param command_list string[] Command list to execute.
----@param opts table
+--- One shot jump
```
Comment (elianiva) [inline lua/telescope/finders.lua:150]: is it `jump` or `job` ?

File: lua/telescope/sorters.lua [hunk 2]
```diff
@@ -32,12 +32,17 @@ Sorter.__index = Sorter
 ---
 --- Lower number is better (because it's like a closer match)
 --- But, any number below 0 means you want that line filtered out.
---- @field scoring_function function Function that has the interface:
---      (sorter, prompt, line): number
+---@field scoring_function function: Function that has the interface: (sorter, prompt, line): number
+---@field tags table: I actually don't know :)
```
Comment (fdschmidt93) [inline lua/telescope/sorters.lua:36]: We can maybe chime this in for tidiness :)
```suggestion
---@field tags table: Unique tags collected at filtering for tag completion
```
`tags` was introduced to enable #626

File: lua/telescope/finders/async_job_finder.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      args = command_list,
+    }
+  end
+
+  local job
+  return setmetatable({
+    close = function() end,
+  }, {
+    __call = void(async(function(prompt, process_result, process_complete)
+      print("are we callin anything?", job)
```
Comment (tami5) [inline lua/telescope/finders/async_job_finder.lua:32]: debug?

File: lua/telescope/finders/async_job_finder.lua [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        maximum_results = opts.maximum_results,
+        writer = writer,
+        enable_recording = false,
+
+        on_stdout = vim.schedule_wrap(function(_, line)
+          if not line or line == "" then
+            return
+          end
+
+          -- TODO: shutdown job here.
```
Comment (tami5) [inline lua/telescope/finders/async_job_finder.lua:61]: still relevant?

**Review by tami5 — APPROVED** [review]

File: lua/telescope/finders/async_job_finder.lua (hunk 3 above)
Comment (tjdevries) [reply lua/telescope/finders/async_job_finder.lua:32]: This code isn't called anywhere at the moment. async job finder is not enabled yet.

**Comment by dagadbm** [comment]
just tested this.
NOW i can use telescope :D only think missing is  the  optional preview, or at least a better visual feedback that the preview is loading

**Pull request merged**